src/core/net/ip6_filter.cpp

```diff
@@ -124,6 +124,8 @@ otError Filter::AddUnsecurePort(uint16_t aPort)
 {
     otError error = OT_ERROR_NONE;
 
+    VerifyOrExit(aPort != 0, error = OT_ERROR_INVALID_ARGS);
```

Comment:
Assert here and move the check to api file?

Reply:
@bukepo  agree it might be more reasonable. feel free to submit a separate PR to enhance it. thanks.